spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/DecisionTreeRegressor.java

```
@@ -0,0 +1,243 @@
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.FileSet;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.StageConfigurer;
+import co.cask.cdap.etl.api.batch.SparkCompute;
+import co.cask.cdap.etl.api.batch.SparkExecutionPluginContext;
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.Doubles;
+import joptsimple.internal.Strings;
+import org.apache.avro.reflect.Nullable;
+import org.apache.spark.SparkContext;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.mllib.linalg.Vector;
+import org.apache.spark.mllib.linalg.Vectors;
+import org.apache.spark.mllib.tree.model.DecisionTreeModel;
+import org.apache.twill.filesystem.Location;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.ws.rs.Path;
+
+/**
+ * SparkCompute that uses a trained model to tag input records using Decision Tree regression.
+ */
+@Plugin(type = SparkCompute.PLUGIN_TYPE)
+@Name(DecisionTreeRegressor.PLUGIN_NAME)
+@Description("Uses a trained Decision Tree Regression model and regress records.")
+public class DecisionTreeRegressor extends SparkCompute<StructuredRecord, StructuredRecord> {
+  public static final String PLUGIN_NAME = "DecisionTreeRegressor";
+  private DecisionTreeRegressorConfig config;
+  private Schema outputSchema;
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    Schema inputSchema = stageConfigurer.getInputSchema();
+    // if null, the input schema is unknown, or has multiple schemas.
+    if (inputSchema == null) {
+      outputSchema = null;
```

actually, we should throw an exception if the input schema is null.

The reason is that we require a constant input schema, otherwise the record fields might get mapped to different features depending on the order they appear in the schema.